Include/internal/pycore_interp.h

```diff
@@ -102,7 +102,7 @@ struct _is {
        In order to be effective, this must be set to 0 during or right
        after allocation. */
     int _initialized;
-    int finalizing;
+    int started_finalization;  /* set early, used to block some actions. */
```

Comment:
Can you give some examples here of actions which are blocked?

Reply:
Depending upon implementation i'm not sure we even want this flag anymore...  _If_ we allow threading and forking to happen up until the modern `_finalizing` API returns False there'd be no need for it.

Otherwise "some actions" needing describing is basically those two and an odd debug thing in unicodeobject.c as far the users of this go.  Every place that touches it can be seen in this PR.